cpp/CMakeLists.txt

```diff
@@ -56,7 +56,7 @@ set(UCXX_BUILD_TESTS ${BUILD_TESTS})
 set(UCXX_BUILD_BENCHMARKS ${BUILD_BENCHMARKS})
 set(UCXX_BUILD_EXAMPLES ${BUILD_EXAMPLES})
 
-set(UCXX_CXX_FLAGS "")
+set(UCXX_CXX_FLAGS "-Wall" "-Wattributes" "-Werror")
```

Comment:
Other things that might be worthwhile adding `-Wextra`, `-Wsign-conversion`.

Reply:
Thanks Lawrence for the suggestions. I've extended those in https://github.com/rapidsai/ucxx/pull/340/commits/ba26e7e65e26324077346a3e3275c3c4368379a7 . Additionally I added `-Wno-missing-field-initializers` this is because we otherwise hit those warnings when initializing many of the UCP structures where we don't (and probably shouldn't, at least not in all cases) fill those values. Alternatively we could use `#pragma`s to filter them out, but I'm not sure if it's worth it. If you have other better ideas please let me know as well.